### PR TITLE
Extract common payment method operations

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Extracts some common code between PaymentSheet and PaymentSheetFlow controller w.r.t to modify payment methods.
- Improves overall testing for these code paths.

## Motivation
- Reuse for vertical mode

## Testing
- Existing tests
- New tests

## Changelog
N/A

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetFlowControllerViewController.swift [hunk 1]
```diff
@@ -541,9 +545,9 @@ extension PaymentSheetFlowControllerViewController: SavedPaymentOptionsViewContr
             throw PaymentSheetError.unknown(debugDescription: "Failed to read ephemeral key secret")
         }
 
-        return try await configuration.apiClient.updatePaymentMethod(with: paymentMethod.stripeId,
-                                                                     paymentMethodUpdateParams: updateParams,
-                                                                     ephemeralKeySecret: ephemeralKey)
+        return try await savedPaymentMethodManager.update(paymentMethod: paymentMethod,
+                                                          with: updateParams,
+                                                          using: ephemeralKey)
     }
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetFlowControllerViewController.swift:551]: @wooj-stripe why doesn't this deal with CustomerSession?

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/SavedPaymentMethodManagerTest.swift [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    func stubUpdatePaymentMethod(paymentMethod: STPPaymentMethod, ephemeralKey: String) -> XCTestExpectation {
+        return stubRequest(urlContains: "/payment_methods/\(paymentMethod.stripeId)",
+                           ephemeralKey: ephemeralKey,
+                           httpMethod: "POST",
+                           responseObject: STPPaymentMethod.paymentMethodJson)
+    }
+
+    func stubDetachPaymentMethod(paymentMethod: STPPaymentMethod, ephemeralKey: String) -> XCTestExpectation {
+        return stubRequest(urlContains: "/payment_methods/\(paymentMethod.stripeId)/detach",
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/SavedPaymentMethodManagerTest.swift:117]: Does this assert that the method was called?

Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/SavedPaymentMethodManagerTest.swift:117]: Not directly but yes, we return the `XCTestExpectation` which is fulfilled [here](18eadf663dff549cbe31cd38aa433b14643a5bc491b64e40483e0701f85a2c00R99), and each test case waits for the expectations.

Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/SavedPaymentMethodManagerTest.swift:117]: Ah I see now, perfect thanks.

**Review by yuki-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetFlowControllerViewController.swift (hunk 1 above)
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetFlowControllerViewController.swift:551]: Will follow up with @wooj-stripe on this and address in a following PR if needed.

**Pull request merged**